test/e2e/reconciler/namespace/controller_test.go

```diff
@@ -146,6 +149,39 @@ func TestNamespaceScheduler(t *testing.T) {
 				err = server.expect(namespace, unscheduledMatcher(nscontroller.NamespaceReasonSchedulingDisabled))
 				require.NoError(t, err, "did not see namespace marked with scheduling disabled")
 			},
+		}, {
+			name: "enabling scheduling for a workspace prompts namespace reconciliation",
+			work: func(ctx context.Context, t *testing.T, f *framework.KcpFixture, server runningServer) {
+				t.Log("Disable scheduling for the containing workspace")
+				framework.SetWorkspaceSchedulable(t, server.orgKcpClient, server.clusterName, false)
+
+				t.Log("Create a namespace when workspace scheduling is disabled and expect it to remain unscheduled with no condition")
+				namespace, err := server.client.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "e2e-nss-",
+					},
+				}, metav1.CreateOptions{})
+				require.NoError(t, err, "failed to create namespace")
+
+				// Give the namespace scheduler time to reconcile the namespace
+				time.Sleep(5 * time.Second)
```

Comment:
can't we poll for it?

Reply:
Updated to poll for the lack of a change, PTAL.